packages/openapi-v3/src/extension-point/types.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/openapi-v3
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {BindingTemplate} from '@loopback/context';
+import {extensionFor} from '@loopback/core';
+import {OpenApiSpec} from '../types';
+
+/**
+ * Typically an extension point defines an interface as the contract for
+ * extensions to implement
+ */
+export interface OAISpecContributor {
+  /**
+   * The name of root level fields in an OpenAPI specification,
+   * e.g. ROOT(the entire spec), info, paths, security, tags, etc...
+   *
+   *
+   * // question: should we use a union type to enum the valid fields
+   * // instead of using string?
+   */
+  fieldName: string;
```

Comment:
I don't think we need to have `fieldName` for filtering. Each contributor will have the opportunity to mutate the spec.

Reply:
then the implementation should be simpler, I don't have a very strong opinion on it and am glad to go with the simpler solution, it also gives a better flexibility.